python_modules/dagster/dagster/_core/definitions/freshness_checks/utils.py

```diff
@@ -146,16 +159,198 @@ def get_last_updated_timestamp(record: Optional[EventLogRecord]) -> Optional[flo
 
 
 def ensure_freshness_checks(checks: Sequence[AssetChecksDefinition]) -> None:
-    from .non_partitioned import NON_PARTITIONED_FRESHNESS_PARAMS_METADATA_KEY
-    from .time_window_partitioned import TIME_WINDOW_PARTITIONED_FRESHNESS_PARAMS_METADATA_KEY
-
-    for asset_check in checks:
-        for check_spec in asset_check.check_specs:
+    for check_def in checks:
+        for check_spec in check_def.check_specs:
             check.invariant(
-                check_spec.metadata
-                and (
-                    NON_PARTITIONED_FRESHNESS_PARAMS_METADATA_KEY in check_spec.metadata
-                    or TIME_WINDOW_PARTITIONED_FRESHNESS_PARAMS_METADATA_KEY in check_spec.metadata
-                ),
+                check_spec.metadata and check_spec.metadata.get(FRESHNESS_PARAMS_METADATA_KEY),
                 f"Asset check {check_spec.key} didn't have expected metadata. Please ensure that the asset check is a freshness check.",
             )
+
+
+def get_last_update_time_lower_bound(
+    deadline_cron: Optional[str],
+    current_timestamp: float,
+    lower_bound_delta: datetime.timedelta,
+    freshness_cron_timezone: str,
```

Comment:
this should be `deadline_cron_timezone`, right?

Reply:
Or should it actually just be`timezone`, because it also applies to the `timedelta` param?

Reply:
it should just be timezone i guess. All the relevant math is done in terms of this timezone and the result is given in terms of this timezone, I think it's fair to just call it timezone.

Reply:
👍